builder/docker.go

```diff
@@ -164,20 +165,37 @@ func (b *Builder) getDockerMounts() ([]string, error) {
 // image capable of running the desired command, build that image, and then
 // run the command in that image.
 func (b *Builder) RunCommandInContainer(cmd []string) error {
-	format, err := b.getUpstreamFormat(b.UpstreamVer)
-	if err != nil {
-		return err
+	var imageName string
+	var format string
+	var err error
+	if !Offline {
+		format, err = b.getUpstreamFormat(b.UpstreamVer)
+		if err != nil {
+			return err
+		}
+	} else {
+		if err = helpers.RunCommandSilent("docker", "image", "inspect", "clearlinux/mixer:"+b.State.Mix.Format); err != nil {
+			return errors.Errorf("Failed to find usable docker image, cannot run offline: %s", err)
+		}
+		var hostFormat []byte
+		hostFormat, err = ioutil.ReadFile("/usr/share/defaults/swupd/format")
+		if err != nil {
+			return err
+		}
+		format = string(hostFormat)
+
 	}
 
-	imageName, err := b.getDockerImageName(format)
+	imageName, err = b.getDockerImageName(format)
```

Comment:
Given the comment above, in this point getDockerImageName() will fail if you changed the format from `b.State.Mix.Format` to `/usr/share/defaults/swupd/format`. If you really need to enforce the use of the host format, it may be better to set it before the call to docker image inspect, so you can fail earlier.